sphinx/source/docs/dev_guide/writing_tests.rst

```diff
@@ -0,0 +1,333 @@
+:ref:`devguide_testing_local_typescript` for information on how to run them.
+
+Information on contributing to BokehJS is available in
+:ref:`devguide_bokehjs`.
+
+.. _devguide_writing_tests_bokehjs_unit:
+
+BokehJS unit tests
+~~~~~~~~~~~~~~~~~~
+
+The :term:`BokehJS` unit tests help make sure that the individual sections of
+BokehJS function as expected. The unit tests for BokehJS are located in the
+:bokeh-tree:`bokehjs/test/unit/` folder and sub-folders.
+
+Use `Chai "expect" assertion style <Chai>`_ when writing unit tests for the
+BokehJS testing framework. See the `API documentation of the Chai Assertion
+Library <Chai documentation>`_ for more details on this style.
+
+Use ``expect()`` together with the following elements to create assertions for
+the BokehJS testing framework:
+
+* ``to`` and ``be``: tokens to improve readability of assertions and connect
+  elements
+* ``not``: negates the following assertions
+* ``throw``: asserts that an error is thrown. Accepts the following optional
+  parameters: ``error_type`` (filter by ``Error``) and ``pattern`` (filter by
+  regular expression or string).
+* ``equal``: asserts strict equality (``===``). Expects an operand to compare
+  to.
+* ``similar``: asserts similarity within a defined tolerance. Expects an operand
+  to compare to as well as an optional ``number`` as ``tolerance``.
+* ``identical``: asserts same-value equality. Expects an operand to compare
+  to.
+* ``instanceof``: asserts that the tested element is an instance of the given
+  constructor. Expects a ``Constructor`` to test against.
+* ``undefined``: asserts strict equality (``===``) to ``undefined``
+* ``null``: asserts strict equality (``===``) to ``null``
+* ``true``: asserts strict equality (``===``) to ``true``
+* ``false``: asserts strict equality (``===``) to ``false``
+* ``NaN``: asserts that the tested element is ``NaN``
+* ``empty``: asserts a length of ``0`` (for example, an empty string or an
+  iterable that does not contain any retrievable values)
+* ``below``: asserts that the tested element is below (``<``) a value. Expects a
+  ``number`` to compare to.
+* ``above``: asserts that the tested element is below (``>``) a value. Expects a
+  ``number`` to compare to.
+
+Some examples:
+
+.. code-block:: TypeScript
+
+    expect(m.name).to.be.null
+    expect(grid0).to.be.instanceof(Column)
+    expect(h.msgid).to.not.be.equal(h2.msgid)
+
+In addition to ``expect()``, the BokehJS testing framework also uses an
+``expect_element()`` function. You can use this function in combination with
+``have`` and ``equal_attributes`` to test whether an element has certain
+attributes. ``equal_attributes`` accepts an array of strings as an optional
+``ignored_attributes`` parameter.
+
+For example:
+
+.. code-block:: TypeScript
+
+    expect_element(svg).to.have.equal_attributes(string_to_html(`<p>test</p>`))
+
+If you add new test files, you should add an entry in the directory index file.
```

Comment:
@mattpap Could you provide some background here? I was not able to find a file named `ìndex` in `bokehjs/tests/unit` folder. Or is it possible that the index file refers to non-unit tests for BokehJS, such as defaults and codebases tests that do have an `index.ts` file?
For reference, this is the original sentence in the current docs: https://docs.bokeh.org/en/latest/docs/dev_guide/testing.html#javascript-unit-tests